tests/map_test.go

```diff
@@ -100,6 +101,96 @@ func Test_MapRange(t *testing.T) {
 	}
 }
 
+func Test_MapWrapper(t *testing.T) {
+	// This tests that various map types, and a map as a function argument and return,
+	// wrap and unwrap correctly.
+	type Dummy struct {
+		Msg string
+	}
+
+	type StructWithMap struct {
+		StringMap map[string]string
+		IntMap    map[int]int
+		DummyMap  map[string]*Dummy
+		MapFunc   func(map[string]string) map[string]string
+	}
+
+	dummyMap := map[string]*Dummy{`key`: {Msg: `value`}}
+	swm := &StructWithMap{
+		StringMap: map[string]string{`key`: `value`},
+		IntMap:    map[int]int{1: 2},
+		DummyMap:  dummyMap,
+		MapFunc: func(m map[string]string) map[string]string {
+			return m
+		},
+	}
+	swmWrapper := js.MakeFullWrapper(swm)
+	swmUnwrapped := swmWrapper.Interface().(*StructWithMap)
+	mapFuncArg := map[string]string{`key2`: `value2`}
+
+	if swmWrapper.Get(`StringMap`).Get(`key`).String() != swm.StringMap[`key`] {
+		t.Error(`StringMap did not match`)
+	}
+	if swmWrapper.Get(`IntMap`).Get(`1`).Int() != swm.IntMap[1] {
+		t.Error(`IntMap did not match`)
+
+	}
+	if swmWrapper.Get(`DummyMap`).Get(`key`).Get(`Msg`).String() != swm.DummyMap[`key`].Msg {
+		t.Error(`DummyMap did not match`)
+
+	}
+	if swmWrapper.Call(`MapFunc`, mapFuncArg).Get(`key2`).String() != mapFuncArg[`key2`] {
+		t.Error(`MapFunc did not match`)
+
+	}
+
+	if swmUnwrapped.StringMap[`key`] != swm.StringMap[`key`] {
+		t.Error(`Unwrapped StringMap did not match`)
+	}
+	if swmUnwrapped.IntMap[1] != swm.IntMap[1] {
+		t.Error(`Unwrapped IntMap did not match`)
+	}
+	if swmUnwrapped.DummyMap[`key`].Msg != swm.DummyMap[`key`].Msg {
+		t.Error(`Unwrapped DummyMap did not match`)
+	}
+	if swmUnwrapped.MapFunc(mapFuncArg)[`key2`] != swm.MapFunc(mapFuncArg)[`key2`] {
+		t.Error(`Unwrapped MapFunc did not match`)
+	}
+}
+
+func Test_MapStructObjectWrapper(t *testing.T) {
```

Comment:
When I was talking about embedding `*js.Object` I was referring to something like this:

```go
o := js.Global.Get("JSON").Call("parse", `{"props": {"one": 1, "two": 2}}`)

type data struct {
  *js.Object
  Props map[string]int `js:"props"`
}

d := data{Object: o}
d.props["one"] // Expect: 1
d.props["two"] // Expect: 2
```

Reply:
Oh, okay, I haven't done much with that before.  Hopefully the test I cribbed from your example is sufficient to show correctness?